src/appengine/libs/issue_management/github.py

```diff
@@ -0,0 +1,167 @@
+  """Generate the closing comment of the issue"""
+  return GithubIssueCloseCommentText.format(
+      bug_information=testcase.bug_information)
+
+
+def get_github_access():
+  """Get access to GitHub with the oss-fuzz personal access token"""
+  token = db_config.get_value("oss_fuzz_robot_github_personal_access_token")
+  if not token:
+    logs.log_error("Unable to get oss-fuzz-robot's personal access token.")
+    return None
+  return github.Github(token)
+
+
+def github_filing_enabled(testcase):
+  """Check if the project YAML file requires to file a github issue."""
+  require_github_issue = data_handler.get_value_from_job_definition(
+      testcase.job_type, 'FILE_GITHUB_ISSUE', default='False')
+  return require_github_issue.lower() == 'true'
+
+
+def get_github_repo(testcase, github_access):
+  """Get the GitHub repository to file the issue"""
+  github_repo_url = data_handler.get_value_from_job_definition(
+      testcase.job_type, 'MAIN_REPO', '')
+  if not github_repo_url:
+    logs.log_error("Unable to fetch the MAIN_REPO URL from job definition.")
+    return None
+  github_repo_name = github_repo_url.removeprefix('https://github.com/')
+
+  try:
+    target_repo = github_access.get_repo(github_repo_name)
+  except github.UnknownObjectException as e:
+    logs.log_error(f"Unable to locate GitHub repository "
+                   f"named {github_repo_name} from URL: {github_repo_url}.")
+    target_repo = None
+  return target_repo
+
+
+def file_issue_to_github(github_repo, testcase):
+  """Post the issue to the Github repo of the project."""
+  github_issue_title = get_github_issue_title(testcase)
+  github_issue_body = get_github_issue_body(testcase)
+  return github_repo.create_issue(
+      title=github_issue_title, body=github_issue_body)
+
+
+def update_testcase_properties(testcase, github_repo, github_issue):
+  """Update the github-related properties in the FiledBug entity."""
+  testcase.github_repo_id = github_repo.id
+  testcase.github_issue_num = github_issue.number
+
+
+def file_github_issue(testcase):
+  """File a github issue to the GitHub repo of the project"""
+  if not github_filing_enabled(testcase):
+    return
+
+  github_access = get_github_access()
+  if not github_access:
+    logs.log_error("Unable to access github account and file the issue.")
+    return
+  github_repo = get_github_repo(testcase, github_access)
+  if not github_repo:
+    logs.log_error("Unable to locate github repository and file the issue.")
+    return
+  github_issue = file_issue_to_github(github_repo, testcase)
+  update_testcase_properties(testcase, github_repo, github_issue)
+
+
+def issue_recorded(testcase):
+  """Verify the issue has been filed."""
+  return hasattr(testcase, 'github_repo_id') \
+         and testcase.github_repo_id is not None \
+         and hasattr(testcase, 'github_issue_num') \
+         and testcase.github_issue_num is not None
+
+
+def get_github_issue(testcase, github_access):
```

Comment:
nit re naming: We can just call this `get_issue` to avoid code stutter. 

so callers will call `github.get_issue` instead of the repetitive `github.get_github_issue`. Same with other functions in this file, e.g. `file_github_issue` and more.

Reply:
Ah, that makes sense.
Fixing now, thanks!